io.go

```diff
@@ -45,6 +45,9 @@ func WriteLines(writer io.Writer) Filter {
 				return err
 			}
 			arg.Out <- s
+			if err := b.Flush(); err != nil {
```

Comment:
How about moving the Flush() call before the "arg.Out <- s".  This way, if some downstream processing gets stuck, the user will see the debugging output of the record that caused the processing to get stuck first.